src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/HttpGoogleServicesDAO.scala

```diff
@@ -324,6 +324,31 @@ class HttpGoogleServicesDAO(
     }
   }
 
+  override def getBucketUsage(bucketName: String, maxResults: Option[Long]): Future[BigInt] = {
+    def sumItemSizes(items: java.util.List[StorageObject]) = {
+      (BigInt(0) /: items) { (sum, o) => sum + o.getSize }
+    }
+
+    def processPage(pageToken: Option[String] = None): Future[BigInt] = {
+      val fetcher = getStorage(getBucketServiceAccountCredential).objects().list(bucketName)
+      if (maxResults.isDefined) fetcher setMaxResults maxResults.get
```

Comment:
I'm not 100% what these 2 lines are doing but I think you can roll with a `foreach` 

If you haven't seen it, http://blog.originate.com/blog/2014/06/15/idiomatic-scala-your-options-do-not-match/ has a phenomenal cheatsheet for options